### PR TITLE
Post-anchor-migration clean-up

### PR DESCRIPTION
**Pull request by elliotkennedy**
- Remove processor file and refactor test structure
- Add events and return data to instructions
- Use require and err anchor macros where appropriate

### CODE REVIEW

File: programs/hyperplane/src/instructions/test/runner/processor.rs [hunk 1]
```diff
@@ -0,0 +1,904 @@
+use anchor_lang::solana_program::{
```
Comment (oeble) [inline programs/hyperplane/src/instructions/test/runner/processor.rs:1]: Can you add module documentation about what this is? I'm unsure myself. IIUC, you want native tests calling the handlers directly. What is the added value over banks-client tests?

**Review by oeble — APPROVED** [review]
Nice! only minor stuff

**Review by y2kappa — APPROVED** [review]
lgtm

File: programs/hyperplane/src/instructions/test/runner/processor.rs (hunk 1 above)
Comment (elliotkennedy) [reply programs/hyperplane/src/instructions/test/runner/processor.rs:1]: I don't know if there is any advantage or disadvantage. These are the tests that were in the program and I wouldn't want to migrate the tests at the same time as changing program the framework.

In terms of what it does, it calls the program entrypoint with SVM syscalls stubbed out. It's super quick compared to bpf tests but not sure about banks-client tests.

File: programs/hyperplane/src/instructions/deposit_all_token_types.rs [hunk 2]
```diff
@@ -51,37 +52,37 @@ pub fn handler(
...
-    }
-    let token_b_amount = to_u64(results.token_b_amount)?;
-    if token_b_amount > maximum_token_b_amount {
-        msg!(
+            token_a_amount, maximum_token_a_amount
+        )
+    );
+    require!(token_a_amount > 0, SwapError::ZeroTradingTokens);
+
+    let token_b_amount = dbg_msg!(to_u64(results.token_b_amount))?;
```
[the remark this answers is not in the document]
Comment (elliotkennedy) [reply programs/hyperplane/src/instructions/deposit_all_token_types.rs:73]: Yes I agree it's not great and would like to remove.

`try_msg` sounds like a good idea, can look into.

File: programs/hyperplane/src/instructions/test/runner/processor.rs (hunk 1 above)
Comment (oeble) [reply programs/hyperplane/src/instructions/test/runner/processor.rs:1]: All good. Bpf-test and banks client tests are basically the same thing.

File: programs/hyperplane/src/utils/macros.rs [hunk 3]
```diff
@@ -35,3 +36,55 @@ macro_rules! curve {
...
+        ($($crate::dbg_msg!($val)),+,)
+    };
+}
+
+/// Macro to emit an event and return it from the program
+#[macro_export]
+macro_rules! emitted {
+    ($event: expr) => {
+        anchor_lang::prelude::emit!($event);
+        return Ok($event);
```
Comment (andreihrs) [inline programs/hyperplane/src/utils/macros.rs:88]: out of curiosity, does the TS function return the event?

**Pull request merged**